ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmitterInfo.scala

```diff
@@ -3,35 +3,71 @@
 
 package com.daml.ledger.participant.state.v1
 
-import java.time.Instant
+import com.google.protobuf.ByteString
 
 /** Information provided by the submitter of changes submitted to the ledger.
   *
   * Note that this is used for party-originating changes only. They are
   * usually issued via the Ledger API.
   *
-  * @param actAs: the non-empty set of parties that submitted the change.
+  * @param actAs the non-empty set of parties that submitted the change.
   *
-  * @param applicationId: an identifier for the Daml application that
-  *   submitted the command. This is used for monitoring and to allow Daml
-  *   applications subscribe to their own submissions only.
+  * @param applicationId an identifier for the Daml application that
+  *   submitted the command. This is used for monitoring, command deduplication,
+  *   and to allow Daml applications subscribe to their own submissions only.
   *
-  * @param commandId: a submitter provided identifier that he can use to
-  *   correlate the stream of changes to the participant state with the
-  *   changes he submitted.
+  * @param commandId a submitter-provided identifier to identify an intended ledger change
+  *   within all the submissions by the same parties and application.
   *
-  * @param deduplicateUntil: the time until which the command should be deduplicated.
-  *   Command deduplication is already performed by the participant.
-  *   The ledger may choose to perform additional (cross-participant)
-  *   command deduplication. If it chooses to do so, it must follow the
-  *   same rules as the participant:
-  *   - Deduplication is based on the (submitter, commandId) tuple.
-  *   - Commands must not be deduplicated after the deduplicateUntil time has passed.
-  *   - Commands should not be deduplicated after the command was rejected.
+  * @param deduplicationPeriod The deduplication period for the command submission.
+  *   Used for the deduplication guarantee described in the [[ReadService.stateUpdates]].
+  *
+  * @param submissionId An identifier for the submission that allows an application
+  *   to correlate completions to its submissions.
+  *
+  * @param submissionRank The rank of the submission among all submissions with the same change ID.
+  *   Used for the submission rank guarantee described in the [[ReadService.stateUpdates]].
   */
 final case class SubmitterInfo(
     actAs: List[Party],
     applicationId: ApplicationId,
     commandId: CommandId,
-    deduplicateUntil: Instant,
-)
+    deduplicationPeriod: DeduplicationPeriod,
```

Comment:
@andreaslochbihler-da : the `SubmitterInfo` is also used on `TransactionUpdated`. Would you expect this field to be populated there?

I currently would not, which hints at moving the `deduplicationPeriod` into either the `submitTransaction` method itself or into its own case class.

Reply:
I don't have a strong opinion on that. `deduplicateUntil` was already there and I didn't see a reason to change this now, as I don't know whether the indexer makes use of this field or there are plans to do so. If there are none, then we can pass this as another parameter to `submitTransaction` and `rejectSubmission` instead of making it part of `SubmitterInfo`.

Reply:
Let's remove `deduplicationPeriod` from here, and add it directly to the `WriteService` call

Reply:
I'm not yet convinced. Let's discuss this.

Reply:
As discussed, let's leave it in `SubmitterInfo` now that we've also got `CompletionInfo`.